pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -659,13 +699,22 @@ func (ds *DiscoveryService) ListListeners(request *restful.Request, response *re
 			errorResponse(methodName, response, http.StatusInternalServerError, "LDS "+err.Error())
 			return
 		}
+
+		// TODO: This does not take into account listeners added by webhook
 		resourceCount = uint32(len(listeners))
-		if resourceCount > 0 {
-			ds.ldsCache.updateCachedDiscoveryResponse(key, resourceCount, out)
+
+		transformedOutput, err = ds.invokeWebhook(fmt.Sprintf("/v1/listeners/unused/%s", svcNode), out)
```

Comment:
nit: consider moving this above `resourceCount=...` to keep that line and its TODO closer to the use of `resourceCount`.